cpp/core/utils/TaskContext.cc

```diff
@@ -19,20 +19,49 @@
 #include <folly/Likely.h>
 #include "utils/exception.h"
 
-namespace gluten {
+#include <iostream>
+#include <mutex>
+
+namespace {
+class TaskContextStorage {
+ public:
+  virtual ~TaskContextStorage() {
+    for (auto itr = objects_.rbegin(); itr != objects_.rend(); itr++) {
```

Comment:
do we need this for the loop?
it will be destroyed auto after TaskContextStorage deconstructing.

Reply:
The loop has to be here because we expect the objects to be destroyed in reversed order. Seems to me that `std::vector`'s destructor doesn't guarantee that.

Reply:
OK